CMakeLists.txt

```diff
@@ -298,13 +316,15 @@ add_dependencies(ufs fms)
 if(DATM)
   list(APPEND _ufs_defs_private FRONT_DATM=datm)
   add_dependencies(ufs datatm)
-else()
+elseif(NOT CDEPS_DATM)
```

Comment:
Move dependencies from below to this section. Either keep the dependency section completely separate from the linking section or combine the entire sections.
- [ ] 346
- [ ] 360
- [ ] 375

Reply:
I just follow their convention. The existing build was defining them in two operate place. Of course, it will be more efficient with your suggestion but I would rather to keep the convention same. Also, the build system requires major work to make it more flexible and generic.